drivers/SmartThings/matter-switch/src/third-reality/init.lua

```diff
@@ -0,0 +1,410 @@
+    device:set_field(RECEIVED_Y, nil)
+  end
+end
+
+local function y_attr_handler(driver, device, ib, response)
+  local x = device:get_field(RECEIVED_X)
+  if x == nil then
+    device:set_field(RECEIVED_Y, ib.data.value)
+  else
+    local y = ib.data.value
+    local h, s, _ = color_utils.safe_xy_to_hsv(x, y)
+    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorControl.hue(h))
+    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorControl.saturation(s))
+    device:set_field(RECEIVED_X, nil)
+  end
+end
+
+--TODO setup configure handler to read this attribute.
+local function color_cap_attr_handler(driver, device, ib, response)
+  if ib.data.value ~= nil then
+    if ib.data.value & 0x1 then
+      device:set_field(HUESAT_SUPPORT, true)
+    end
+  end
+end
+
+local third_reality_handler = {
+  NAME = "Third Reality Handler",
+  lifecycle_handlers = {
+    init = device_init,
+    removed = device_removed
+  },
+  matter_handlers = {
+    attr = {
+      [clusters.OnOff.ID] = {
+        [clusters.OnOff.attributes.OnOff.ID] = on_off_attr_handler,
+      },
+      [clusters.LevelControl.ID] = {
+        [clusters.LevelControl.attributes.CurrentLevel.ID] = level_attr_handler
+      },
+      [clusters.ColorControl.ID] = {
+        [clusters.ColorControl.attributes.CurrentHue.ID] = hue_attr_handler,
+        [clusters.ColorControl.attributes.CurrentSaturation.ID] = sat_attr_handler,
+        [clusters.ColorControl.attributes.ColorTemperatureMireds.ID] = temp_attr_handler,
+        [clusters.ColorControl.attributes.CurrentX.ID] = x_attr_handler,
+        [clusters.ColorControl.attributes.CurrentY.ID] = y_attr_handler,
+        [clusters.ColorControl.attributes.ColorCapabilities.ID] = color_cap_attr_handler,
+      },
+      [clusters.IlluminanceMeasurement.ID] = {
+        [clusters.IlluminanceMeasurement.attributes.MeasuredValue.ID] = illuminance_attr_handler
+      },
+      [clusters.OccupancySensing.ID] = {
+        [clusters.OccupancySensing.attributes.Occupancy.ID] = occupancy_attr_handler,
+      }
+    },
+    fallback = matter_handler,
+  },
+  subscribed_attributes = {
+    [capabilities.switch.ID] = {
+      clusters.OnOff.attributes.OnOff
+    },
+    [capabilities.switchLevel.ID] = {
+      clusters.LevelControl.attributes.CurrentLevel
+    },
+    [capabilities.colorControl.ID] = {
+      clusters.ColorControl.attributes.CurrentHue,
+      clusters.ColorControl.attributes.CurrentSaturation,
+      clusters.ColorControl.attributes.CurrentX,
+      clusters.ColorControl.attributes.CurrentY,
+    },
+    [capabilities.colorTemperature.ID] = {
+      clusters.ColorControl.attributes.ColorTemperatureMireds,
+    },
+    [capabilities.illuminanceMeasurement.ID] = {
+      clusters.IlluminanceMeasurement.attributes.MeasuredValue
+    },
+    [capabilities.motionSensor.ID] = {
+      clusters.OccupancySensing.attributes.Occupancy
+    }
+  },
+  capability_handlers = {
```

Comment:
You can remove all of these `capability_handlers`. Since these functions are identical to the same functions in the parent driver, we can remove these so that the driver will default to using these functions in the parent driver (`matter-switch`)